unit-tests/run-unit-tests.py

```diff
@@ -369,29 +382,80 @@ def is_live( self ):
         """
         Returns True if the test configurations specify devices (test has a 'device' directive)
         """
-        return self._config and len(self._config.configurations) > 0
+        return self._config and len( self._config.configurations ) > 0
+
+    def get_source_path( self ):

```

Comment:
Since this is not a simple `get`, I'd prefer `find` which implies some sort of calculation that might take time